src/memory.rs

```diff
@@ -279,7 +285,7 @@ impl<'a, 'tcx> Memory<'a, 'tcx> {
             Some(alloc) => Ok(alloc),
             None => match self.functions.get(&id) {
                 Some(_) => Err(EvalError::DerefFunctionPointer),
-                None if id == ZST_ALLOC_ID => Err(EvalError::ZstAllocAccess),
+                None if id == ZST_ALLOC_ID => Err(EvalError::InvalidMemoryAccess),
```

Comment:
Is this still needed? I don't see the relationship between `ZST_ALLOC_ID` and... aaaah nevermind :).